processor/transformer/transformer.go

```diff
@@ -125,50 +146,59 @@ func isJobTerminated(status int) bool {
-	config.RegisterIntConfigVariable(5, &maxHTTPIdleConnections, false, 1, "Processor.maxHTTPIdleConnections")
-	config.RegisterBoolConfigVariable(true, &disableKeepAlives, false, "Transformer.Client.disableKeepAlives")
-
-	config.RegisterIntConfigVariable(30, &maxRetry, true, 1, "Processor.maxRetry")
-	config.RegisterDurationConfigVariable(100, &retrySleep, true, time.Millisecond, []string{"Processor.retrySleep", "Processor.retrySleepInMS"}...)
-	config.RegisterDurationConfigVariable(30, &timeoutDuration, false, time.Second, "HttpClient.procTransformer.timeout")
-}
-
 type TransformerResponseT struct {
 	// Not marking this Singular Event, since this not a RudderEvent
 	Output           map[string]interface{} `json:"output"`
 	Metadata         MetadataT              `json:"metadata"`
 	StatusCode       int                    `json:"statusCode"`
 	Error            string                 `json:"error"`
-	ValidationErrors []ValidationErrorT     `json:"validationErrors"`
+	ValidationErrors []ValidationError      `json:"validationErrors"`
 }
 
-type ValidationErrorT struct {
+type ValidationError struct {
 	Type    string            `json:"type"`
 	Message string            `json:"message"`
 	Meta    map[string]string `json:"meta"`
 }
 
 // Setup initializes this class
-func (trans *HandleT) Setup() {
-	trans.logger = pkgLogger
-	trans.sentStat = stats.Default.NewStat("processor.transformer_sent", stats.CountType)
-	trans.receivedStat = stats.Default.NewStat("processor.transformer_received", stats.CountType)
-	trans.cpDownGauge = stats.Default.NewStat("processor.control_plane_down", stats.GaugeType)
+func (trans *HandleT) Setup(conf *config.Config, log logger.Logger, stat stats.Stats) {
+	trans.conf = conf
+	trans.logger = log.Child("transformer")
+	trans.stat = stat
+
+	trans.sentStat = stat.NewStat("processor.transformer_sent", stats.CountType)
+	trans.receivedStat = stat.NewStat("processor.transformer_received", stats.CountType)
+	trans.cpDownGauge = stat.NewStat("processor.control_plane_down", stats.GaugeType)
+
+	trans.config.maxConcurrency = conf.GetInt("Processor.maxConcurrency", 200)
+	trans.config.maxHTTPConnections = conf.GetInt("Processor.maxHTTPConnections", 100)
+	trans.config.maxHTTPIdleConnections = conf.GetInt("Processor.maxHTTPIdleConnections", 5)
+	trans.config.disableKeepAlives = conf.GetBool("Transformer.Client.disableKeepAlives", true)
+	trans.config.timeoutDuration = conf.GetDuration("HttpClient.procTransformer.timeout", 30, time.Second)
+
+	trans.config.destTransformationURL = conf.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
+	trans.config.userTransformationURL = conf.GetString("USER_TRANSFORM_URL", trans.config.destTransformationURL)
 
-	trans.guardConcurrency = make(chan struct{}, maxConcurrency)
+	conf.RegisterIntConfigVariable(30, &trans.config.maxRetry, true, 1, "Processor.maxRetry")
+	conf.RegisterBoolConfigVariable(false, &trans.config.failOnTimeout, true, "Processor.Transformer.failOnTimeout")
```

Comment:
Let's make it more explicit and set this configuration override on hosted rudderstack object already
```suggestion
	conf.RegisterBoolConfigVariable(false, &trans.config.failOnUserTransformTimeout, true, "Processor.Transformer.failOnUserTransformTimeout")
```